Build failed on latest

```
diff --git a/CMakeLists.txt b/CMakeLists.txt
index e4327e5..b49707a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -9,7 +9,7 @@ project("hdr_histogram")
 ENABLE_TESTING()

 if(UNIX)
-    set(CMAKE_C_FLAGS "-Wall -Wno-unknown-pragmas -Wextra -Wshadow -Winit-self -Wmissing-prototypes -D_GNU_SOURCE")
+    set(CMAKE_C_FLAGS "-std=gnu99 -Wall -Wno-unknown-pragmas -Wextra -Wshadow -Winit-self -Wmissing-prototypes -D_GNU_SOURCE")
     set(CMAKE_C_FLAGS_DEBUG "-O0 -g")
     set(CMAKE_C_FLAGS_RELEASE "-O3 -g")
 endif()
```

I am on a Ubuntu 14.04.03 server, and details are given below:

```
yicwang@yicwang-VM:/tmp$ gcc --version
gcc (Ubuntu 4.8.4-2ubuntu1~14.04) 4.8.4
Copyright (C) 2013 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
yicwang@yicwang-VM:/tmp$ cmake --version
cmake version 2.8.12.2
yicwang@yicwang-VM:/tmp/HdrHistogram_c$ cmake .
-- The C compiler identification is GNU 4.8.4
-- The CXX compiler identification is GNU 4.8.4
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Looking for clock_gettime in rt
-- Looking for clock_gettime in rt - found
-- Configuring done
-- Generating done
-- Build files have been written to: /tmp/HdrHistogram_c
yicwang@yicwang-VM:/tmp/HdrHistogram_c$ make
Scanning dependencies of target hdr_histogram
[  6%] Building C object src/CMakeFiles/hdr_histogram.dir/hdr_histogram.c.o
/tmp/HdrHistogram_c/src/hdr_histogram.c: In function ‘hdr_reset_internal_counters’:
/tmp/HdrHistogram_c/src/hdr_histogram.c:187:5: error: ‘for’ loop initial declarations are only allowed in C99 mode
     for (int i = 0; i < h->counts_len; i++)
     ^
/tmp/HdrHistogram_c/src/hdr_histogram.c:187:5: note: use option -std=c99 or -std=gnu99 to compile your code
make[2]: *** [src/CMakeFiles/hdr_histogram.dir/hdr_histogram.c.o] Error 1
make[1]: *** [src/CMakeFiles/hdr_histogram.dir/all] Error 2
make: *** [all] Error 2
```

--- a/src/hdr_histogram.c
+++ b/src/hdr_histogram.c
@@ -183,8 +183,9 @@ void hdr_reset_internal_counters(struct hdr_histogram* h)
     int min_non_zero_index = -1;
     int max_index = -1;
     int64_t observed_total_count = 0;
+    int i;
 
-    for (int i = 0; i < h->counts_len; i++)
+    for (i = 0; i < h->counts_len; i++)
     {
         int64_t count_at_index;
 

--- a/src/hdr_histogram_log.c
+++ b/src/hdr_histogram_log.c
@@ -238,7 +238,8 @@ int hdr_encode_compressed(
     }
 
     int data_index = 0;
-    for (int i = 0; i < counts_limit;)
+    int i;
+    for (i = 0; i < counts_limit;)
     {
         int64_t value = h->counts[i];
         i++;
@@ -315,7 +316,8 @@ void hdr_reset_internal_counters(struct hdr_histogram* h);
 
 static void _apply_to_counts_16(struct hdr_histogram* h, const int16_t* counts_data, const int32_t counts_limit)
 {
-    for (int i = 0; i < counts_limit; i++)
+    int i;
+    for (i = 0; i < counts_limit; i++)
     {
         h->counts[i] = be16toh(counts_data[i]);
     }
@@ -323,7 +325,8 @@ static void _apply_to_counts_16(struct hdr_histogram* h, const int16_t* counts_d
 
 static void _apply_to_counts_32(struct hdr_histogram* h, const int32_t* counts_data, const int32_t counts_limit)
 {
-    for (int i = 0; i < counts_limit; i++)
+    int i;
+    for (i = 0; i < counts_limit; i++)
     {
         h->counts[i] = be32toh(counts_data[i]);
     }
@@ -331,7 +334,8 @@ static void _apply_to_counts_32(struct hdr_histogram* h, const int32_t* counts_d
 
 static void _apply_to_counts_64(struct hdr_histogram* h, const int64_t* counts_data, const int32_t counts_limit)
 {
-    for (int i = 0; i < counts_limit; i++)
+    int i;
+    for (i = 0; i < counts_limit; i++)
     {
         h->counts[i] = be64toh(counts_data[i]);
     }

--- a/test/hdr_histogram_log_test.c
+++ b/test/hdr_histogram_log_test.c
@@ -156,7 +156,8 @@ static void load_histograms()
     hdr_alloc(INT64_C(3600) * 1000 * 1000, 3, &raw_histogram);
     hdr_alloc(INT64_C(3600) * 1000 * 1000, 3, &cor_histogram);
 
-    for (int i = 0; i < 10000; i++)
+    int i;
+    for (i = 0; i < 10000; i++)
     {
         hdr_record_value(raw_histogram, 1000);
         hdr_record_corrected_value(cor_histogram, 1000, 10000);
@@ -298,7 +299,8 @@ static char* test_encode_and_decode_compressed_large()
     hdr_init(1, limit, 4, &expected);
     srand(5);
 
-    for (int i = 0; i < 8070; i++)
+    int i;
+    for (i = 0; i < 8070; i++)
     {
         hdr_record_value(expected, rand() % limit);
     }
@@ -633,7 +635,8 @@ static char* test_string_encode_decode()
     struct hdr_histogram *histogram, *hdr_new = NULL;
     hdr_alloc(INT64_C(3600) * 1000 * 1000, 3, &histogram);
 
-    for (int i = 1; i < 100; i++)
+    int i;
+    for (i = 1; i < 100; i++)
     {
         hdr_record_value(histogram, i*i);
     }

--- a/test/hdr_histogram_perf.c
+++ b/test/hdr_histogram_perf.c
@@ -49,10 +49,12 @@ int main()
     setlocale(LC_NUMERIC, "");
     int64_t iterations = 400000000;
 
-    for (int i = 0; i < 100; i++)
+    int i;
+    for (i = 0; i < 100; i++)
     {
+        int64_t j;
         hdr_gettime(&t0);
-        for (int64_t j = 1; j < iterations; j++)
+        for (j = 1; j < iterations; j++)
         {
             hdr_record_value(histogram, j);
         }

